[Jing Jie] iP

jingjie88 opened the pull request:
# Monke 🦍 🦍 🦍 

> "ooga booga" -Monke 🦧 

Monke can help you remember tasks. It's
- ~~Good~~ _VERY GOOD_
- Text-based
- Easy to use

Here's how to use it,
1. Download it from [here](https://github.com/jingjie88/ip/)
2. Use it

Here's some code from `Monke` constructor,
```
    public Monke(String filepath) {
        this.ui = new Ui();
        this.storage = new Storage(filepath);
        try {
            tasks = new TaskList(storage.load());
        } catch (MonkeException e) {
            tasks = new TaskList();
        }
    }
```

Features:
- [x] Managing tasks
- [x] Marking completed tasks
- [ ] GUI (coming soon)

Use Monke **NOW** 🦍 🦍 🦍

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jingjie88 committed:
Add greet and bye

jingjie88 committed:
Add echo

jingjie88 committed:
Add list

jingjie88 committed:
Add mark as done feature

jingjie88 committed:
Add todo, event, deadline features

jingjie88 committed:
Add UI tests

jingjie88 committed:
Add error handling

jingjie88 committed:
Add delete feature

jingjie88 committed:
Add enum

jingjie88 committed:
Fix bugs

jingjie88 committed:
Add save and load task data automatically

jingjie88 committed:
Merge branch 'branch-Level-7'

jingjie88 committed:
Add dates to deadline

jingjie88 committed:
Add handling of times for deadline

jingjie88 committed:
Fix bug where datetime is stored in data incorrectly

jingjie88 committed:
Merge branch 'branch-Level-8'

jingjie88 committed:
Refactor code to use more OOP

jingjie88 committed:
Organize code into packages

jingjie88 committed:
Merge branch 'add-gradle-support'

jingjie88 committed:
Add Gradle

jingjie88 committed:
Add unit tests

jingjie88 committed:
Add jar file

jingjie88 committed:
Add JavaDoc comments

jingjie88 committed:
Change code to fit coding standard

jingjie88 committed:
Add find feature

jingjie88 committed:
Merge branch 'branch-A-JavaDoc'

jingjie88 committed:
Merge branch 'branch-A-CodingStandard'

jingjie88 committed:
Merge branch 'branch-Level-9'

Gabriel4357 commented on src/main/java/monke/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+        return this.tasks.size();
+    }
+
+    public TaskList filter(String substr) {
```
Could use name to explain. What is this filtering?

Gabriel4357 commented on src/main/java/monke/tasks/Task.java:
```diff
@@ -0,0 +1,67 @@
+ * The Task class represents a task in the Monke application.
+ * It is an abstract class and serves as the base class for specific task types.
+ */
+public abstract class Task {
```
Good naming of classes as nouns and functions as verbs. This is seen throughout the code. 👍

Gabriel4357 commented on src/main/java/monke/tasks/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public class Deadline extends Task {
+    /** The deadline date and time of the task. */
+    private LocalDateTime date;
```
Maybe could indicate time as well since it's not just date 

```suggestion
    private LocalDateTime dateTime;
```

Gabriel4357 commented on src/main/java/monke/tasks/Task.java:
```diff
@@ -0,0 +1,67 @@
+        return (isDone ? "X" : " ");
+    }
+
+    public String getDescription() {
```
Good naming for getters and booleans, seen through the code. 👍

Gabriel4357 reviewed (commented):
Excellent naming of variables. Was very difficult finding things to comment on. 🥇

Carlintyj commented on src/main/java/monke/Storage.java:
```diff
@@ -0,0 +1,68 @@
+
+import monke.tasks.Task;
+
+import java.io.File;
```
Hi, are the imports in the correct order?

Carlintyj commented on src/main/java/monke/Parser.java:
```diff
@@ -0,0 +1,223 @@
+import monke.tasks.Task;
+import monke.tasks.Todo;
+
+import java.time.LocalDateTime;
```
Are these imports in the correct order?

Carlintyj commented on src/main/java/monke/Parser.java:
```diff
@@ -0,0 +1,223 @@
+    public static DeadlineCommand parseDeadline(String args) throws MonkeException {
+        String[] tmp = args.split(" /by ", 2);
+        if (tmp.length < 2 || tmp[0].isBlank() || tmp[1].isBlank()) {
+            throw new MonkeException("You must format your deadline like this:\n" +
```
Should the "+" be in a new line?

Carlintyj commented on src/main/java/monke/commands/FindCommand.java:
```diff
@@ -0,0 +1,29 @@
+import monke.Storage;
+import monke.TaskList;
+import monke.Ui;
+import monke.tasks.Task;
```
Is this import unused?

Carlintyj commented on src/main/java/monke/commands/FindCommand.java:
```diff
@@ -0,0 +1,29 @@
+import monke.Ui;
+import monke.tasks.Task;
+
+import java.util.ArrayList;
```
Is this import unused?

Carlintyj reviewed (commented):
LGTM. Just some nits to fix

jingjie88 committed:
Add gui

jingjie88 committed:
Merge branch 'branch-Level-10'

jingjie88 committed:
Add assertions

jingjie88 committed:
Add code comments for parseEvent() method

The method consists of several steps of splitting strings without code comments.

The lack of code comments causes readability issues.

Let's add code comments explaining the steps in the method.

Code comments can explain what the code is doing, making the code readable and easier to maintain.

jingjie88 committed:
Add some code comments

jingjie88 committed:
Merge pull request #1 from jingjie88/branch-A-CodeQuality

Branch a code quality

jingjie88 committed:
Merge branch 'master' into branch-A-Assertions

jingjie88 committed:
Merge pull request #2 from jingjie88/branch-A-Assertions

Add assertions

jingjie88 committed:
Add snooze feature

There is no way for users to postpone deadlines.

Adding a feature to allow users to postpone deadline would make user experience better.

Let's add a SnoozeCommand class to implement the snooze feature to postpone deadlines.

Using the Command class is consistent with the code so far.

jingjie88 committed:
Merge branch 'branch-B-Snooze'

jingjie88 committed:
Fix bug

jingjie88 committed:
Add readme

jingjie88 committed:
Update README.md

jingjie88 committed:
Add Ui.png
